ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -926,6 +926,14 @@ public Optional<Address> contractAddress() {
     return Optional.empty();
   }
 
+  private Bytes toRlp() {
+    return RLP.encode(this::writeTo);
+  }
+
+  public int calculateSize() {
+    return toRlp().size();
+  }
+
```

Comment:
Do you think it makes sense to cache the size on first call, since encoding the transaction every time we want the size could be expensive?

Reply:
Hey Fabio, it totally makes sense. It will never change anyway, right?

Reply:
yes all the fields of the transactions are final, and while checking that, I found that this could be another optimization, the `hash` of the transactions is calculated and cached in the `getHash` method that already does the serialization, and so it could be used to sent the size, or the other way around if the `getSize` is called first.

Reply:
@Gabriel-Trintinalia I have implemented the memoization of the size here https://github.com/hyperledger/besu/pull/4812 if you want to review it